fhirflat/flat2fhir.py

```diff
@@ -130,12 +159,53 @@ def set_datatypes(k, v_dict, klass) -> dict:
             }
 
         data_type = prop[value_type[0]]["type"]
-        data_class = get_fhirtype(data_type)
-        return {"url": k, f"{value_type[0]}": set_datatypes(k, v_dict, data_class)}
+        try:
+            data_class = get_fhirtype(data_type)
+            return {"url": k, f"{value_type[0]}": set_datatypes(k, v_dict, data_class)}
+        except AttributeError:
+            # datatype should be a primitive
+            return {"url": k, f"{value_type[0]}": v_dict[k]}
 
     return {s.split(".", 1)[1]: v_dict[s] for s in v_dict}
 
 
+def find_data_class(data_class: list[BaseModel] | BaseModel, k: str) -> BaseModel:
```

Comment:
Is this for extracting FHIR resource types from embedded references to other resources?

Reply:
This has been stripped out of the `expand_concepts()` function below.
Yes - the JSON schemas created for each FHIR resource by pydantic return the type of each property as a string (e.g. "CodeableConcept"). This function extracts the resource type and finds the appropriate python class to allow stepping through the nested resources.